bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ThingActionsResource.java

```diff
@@ -230,6 +251,76 @@ public Response executeThingAction(@PathParam("thingUID") @Parameter(description
         }
     }
 
+    private @Nullable ConfigDescriptionParameter convertToConfigDescriptionParameter(Input input) {
+        boolean supported = true;
+        ConfigDescriptionParameter.Type parameterType = ConfigDescriptionParameter.Type.TEXT;
+        String defaultValue = null;
+        boolean required = false;
+        String context = null;
+        switch (input.getType()) {
+            case "boolean":
+                defaultValue = "false";
+                required = true;
+            case "java.lang.Boolean":
+                parameterType = ConfigDescriptionParameter.Type.BOOLEAN;
+                break;
+            case "byte":
+            case "short":
+            case "int":
+            case "long":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Byte":
+            case "java.lang.Short":
+            case "java.lang.Integer":
+            case "java.lang.Long":
+                parameterType = ConfigDescriptionParameter.Type.INTEGER;
+                break;
+            case "float":
+            case "double":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Float":
+            case "java.lang.Double":
+                parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                break;
+            case "java.lang.String":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                break;
+            case "java.time.LocalDate":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                context = "date";
+                break;
+            case "java.time.LocalTime":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                context = "time";
+                break;
+            case "java.time.LocalDateTime":
+            case "java.time.ZonedDateTime":
+                parameterType = ConfigDescriptionParameter.Type.TEXT;
+                context = "datetime";
```

Comment:
Same as for the date context, currently not implemented in the UI.

Reply:
datetime is a documented context value:
https://github.com/openhab/openhab-core/blob/47284e5521c5bccb0db5afec1202a0aa453a2f4f/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigDescriptionParameter.java#L322

Reply:
Globally discussed in the other comment about date context.